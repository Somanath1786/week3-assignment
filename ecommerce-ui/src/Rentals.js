import React from 'react';
import RentalCard from './RentalCard'
import { withStyles } from '@material-ui/core/styles';
import { Grid , Button, Card } from '@material-ui/core';
import { cyan } from '@material-ui/core/colors';

const styles = theme => ({    
    background: {
        background: cyan[300],
    },
    button: {
        margin: theme.spacing.unit,
        marginTop : 20,
        marginBottom : 20,
        float: "left",
        position: "relative",
        right:"-1%"
      },
});

class Rentals extends React.Component {

    render() {
        const {classes, rentals, onClick} = this.props;       
        
        const rentalCards = rentals
                            .map((rental, idx) => {
                                return(
                                    <RentalCard rental={rental} key={idx} onClick={onClick(idx)}/>
                                );
                            });
        return(
            <div>
            <Card className={classes.background}>
                <Button color= "secondary" variant="contained" className={classes.button}> Rentals </Button>
                <Grid container spacing={24}>                   
                    {rentalCards}
                </Grid>
                </Card>    
            </div>
        );
    };
}

export default withStyles(styles)(Rentals);