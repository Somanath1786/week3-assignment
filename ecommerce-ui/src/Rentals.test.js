import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from 'enzyme';
import Rentals from './Rentals';

it('shallow renders', () => {
    const wrapper = shallow(<Rentals />);
    expect(wrapper).toMatchSnapshot();
  
    expect(wrapper.props('classes')).toBeDefined();
    expect(wrapper.props('rentals')).toBeDefined();
    expect(wrapper.props('onClick')).toBeDefined();
  
  });