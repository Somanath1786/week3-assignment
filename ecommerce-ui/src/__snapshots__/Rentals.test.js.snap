// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<Rentals
  classes={
    Object {
      "background": "Rentals-background-1",
      "button": "Rentals-button-2",
    }
  }
/>
`;
